test/python/converters/test_circuit_to_instruction.py

```diff
@@ -186,6 +186,23 @@ def test_parameter_map(self):
         self.assertEqual(inst.definition[2][0].params, [gamma, phi, 0])
         self.assertEqual(str(inst.definition[3][0].params[0]), "gamma + phi")
 
+    def test_registerless_classical_bits(self):
+        """Test that conditions on registerless classical bits can be handled during the conversion.
+
+        Regression test of gh-7394."""
+        expected = QuantumCircuit([Qubit(), Clbit()])
+        expected.h(0).c_if(expected.clbits[0], 0)
+        test = circuit_to_instruction(expected)
+
+        self.assertIsInstance(test, Instruction)
+        self.assertIsInstance(test.definition, QuantumCircuit)
+
+        self.assertEqual(len(test.definition.data), 1)
+        test_instruction, _, _ = test.definition.data[0]
+        expected_instruction, _, _ = expected.data[0]
+        self.assertIs(type(test_instruction), type(expected_instruction))
+        self.assertEqual(test_instruction.condition, (test.definition.clbits[0], False))
```

Comment:
```suggestion
        self.assertEqual(test_instruction.condition, (test.definition.clbits[0], 0))
```
Just for clarity, since we're not implicitly converting from int to bool during `circuit_to_instruction`.

Reply:
Thanks. Not sure why I wrote the two differently at all - I never use `False` normally for these!